app/models/goal.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
 
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .task import Task
+
+
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+    def to_dict(self, withRelationship=False):
+        result = {
+            "id": self.id,
+            "title": self.title
+        }
+        if withRelationship:
+            if self.tasks:
+                result['tasks'] = [task.to_dict() for task in self.tasks]
+            else:
+                result['tasks'] = []
```

Comment:
`self.tasks` will be an empty list if a goal does not have any tasks, so we don't need `withRelationship` at all. We can remove it from the function definition too. 

```suggestion
        if self.tasks:
            result['tasks'] = [task.to_dict() for task in self.tasks]
        else:
            result['tasks'] = []
```